osgp/shared/osgp-ws-smartmetering/src/main/resources/schemas/sm-monitoring.xsd

```diff
@@ -540,4 +540,89 @@
     </xsd:sequence>
   </xsd:complexType>
 
+  <!-- ActualPowerQuality -->
+
+  <xsd:element name="ActualPowerQualityRequest">
+    <xsd:complexType>
+      <xsd:complexContent>
+        <xsd:extension base="tns:ActualPowerQualityRequestData">
+          <xsd:sequence>
+            <xsd:element name="DeviceIdentification" type="common:DeviceIdentification"/>
+          </xsd:sequence>
+        </xsd:extension>
+      </xsd:complexContent>
+    </xsd:complexType>
+  </xsd:element>
+
+  <xsd:complexType name="ActualPowerQualityRequestData">
+    <xsd:complexContent>
+      <xsd:extension base="common:Query">
+        <xsd:sequence>
+          <xsd:element name="ProfileType" type="xsd:string"/>
+        </xsd:sequence>
+      </xsd:extension>
+    </xsd:complexContent>
+  </xsd:complexType>
+
+  <xsd:element name="ActualPowerQualityResponse">
+    <xsd:complexType>
+      <xsd:complexContent>
+        <xsd:extension base="tns:ActualPowerQualityResponseData"></xsd:extension>
+      </xsd:complexContent>
+    </xsd:complexType>
+  </xsd:element>
+
+  <xsd:complexType name="ActualPowerQualityResponseData">
+    <xsd:complexContent>
+      <xsd:extension base="common:Response">
+        <xsd:sequence>
+          <xsd:element name="ActualPowerQualityData" type="tns:ActualPowerQualityData"
+                       minOccurs="1"
+                       maxOccurs="1"/>
+        </xsd:sequence>
+      </xsd:extension>
+    </xsd:complexContent>
+  </xsd:complexType>
+
+  <xsd:complexType name="ActualPowerQualityData">
+    <xsd:sequence>
+      <xsd:sequence>
+        <xsd:element name="CaptureObjects" type="common:CaptureObjects" minOccurs="1" maxOccurs="1"/>
+        <xsd:element name="ActualValues" type="tns:ActualValues" minOccurs="1" maxOccurs="1"/>
+      </xsd:sequence>
+    </xsd:sequence>
+  </xsd:complexType>
+
+  <xsd:complexType name="ActualValues">
+    <xsd:sequence>
+      <xsd:element type="tns:ActualValue" name="ActualValue"
+                   minOccurs="0" maxOccurs="unbounded" />
+    </xsd:sequence>
+  </xsd:complexType>
+
+  <xsd:complexType name="ActualValue">
```

Comment:
this doesn't seem OK to me. The object is exactly the same as the PQprofile return data. Why duplicate this?
Instead of duplicating I would reuse the existing model and make that more generic.

Also, ActualValue is too generic IMHO. Should be renamed to PowerQualityValue or something like that.

<xsd:complexType name="ProfileEntryValue">
    <xsd:choice maxOccurs="unbounded">
      <xsd:element name="StringValue" type="xsd:string" />
      <xsd:element name="DateValue" type="xsd:dateTime" />
      <xsd:element name="FloatValue" type="xsd:decimal" />
      <xsd:element name="LongValue" type="xsd:long" />
    </xsd:choice>
  </xsd:complexType>